# This file is covered by the LICENSE file in the root of this project.
ref_related:    
  frames_interval: 30  #you might need to adjust this parameter if you want to use your own dataset
instance_paras:
  voxel_leaf_size: 0.05 
  ins_fuse_radius: 0.5 #radius threshold for instance fusion when concuating
rsn_paras:
  xy_range: 80         #xy limits for road grids
  grid_size: 10        #size of the road grid, e.g., 10m*10m
  mini_p_num: 1000     #minimum number of points in a road grid to be considered for road normal estimation, if you downsample the point cloud (or using other dataset), you may need to adjust this parameter
  sac_threshold: 0.04  #threshold for SACSegmentation
  fuse_radius: 3.5     #radius threshold for road normal fusion when concuating, the smaller, the more normals generated, and the more memory needed
labels: 
  0 : "car"
  1 : "bicycle"
  2 : "motorcycle"
  3 : "truck"
  4 : "other-vehicle"
  5 : "person"
  6 : "bicyclist"
  7 : "motorcyclist"
  8 : "road"
  9 : "parking"
  10: "sidewalk"
  11: "other-ground"
  12: "building"
  13: "fence"
  14: "vegetation"
  15: "trunk"
  16: "terrain"
  17: "pole"
  18: "traffic-sign"
classes_for_graph:        # False: not-used, True:used, 
  0 : False #True  #"car"
  1 : False        #"bicycle"
  2 : False        #"motorcycle"
  3 : False        #"truck"
  4 : False        #"other-vehicle"
  5 : False        #"person"
  6 : False        #"bicyclist"
  7 : False        #"motorcyclist"
  8 : False        #"road"
  9 : False #True  #"parking"
  10: False #True  #"sidewalk"
  11: False        #"other-ground"
  12: False        #"building"
  13: False #True  #"fence"
  14: False #True  #"vegetation"
  15: True         #"trunk"
  16: False        #"terrain"
  17: True         #"pole"
  18: True         #"traffic-sign"
instance_seg_para:     #instance segmentation parameters
  0 : 0.5        #"car"
  1 : 0.1        #"bicycle"
  2 : 0.1        #"motorcycle"
  3 : 0.1        #"truck"
  4 : 0.1        #"other-vehicle"
  5 : 0.5        #"person"
  6 : 0.1        #"bicyclist"
  7 : 0.1        #"motorcyclist"
  8 : 0.1        #"road"
  9 : 0.5        #"parking"
  10: 0.5        #"sidewalk"
  11: 0.1        #"other-ground"
  12: 1.0        #"building"
  13: 0.5        #"fence"
  14: 0.5        #"vegetation"
  15: 0.5        #"trunk"
  16: 0.5        #"terrain"
  17: 0.5        #"pole"
  18: 0.5        #"traffic-sign"
mini_point_one_instance:      #minimum number of points in an instance
  0 : 80         #"car"
  1 : 10         #"bicycle"
  2 : 10         #"motorcycle"
  3 : 10         #"truck"
  4 : 10         #"other-vehicle"
  5 : 10         #"person"
  6 : 10         #"bicyclist"
  7 : 10         #"motorcyclist"
  8 : 10         #"road"
  9 : 80         #"parking"
  10: 200        #"sidewalk"
  11: 10         #"other-ground"
  12: 200        #"building"
  13: 20         #"fence"
  14: 150        #"vegetation"
  15: 20         #"trunk"
  16: 10         #"terrain"
  17: 20         #"pole"
  18: 20         #"traffic-sign"
color_map: # bgr
  0 : [245, 150, 100]  #"car"
  1 : [245, 230, 100]  #"bicycle"
  2 : [150, 60, 30]    #"motorcycle"
  3 : [180, 30, 80]    #"truck"
  4 : [255, 0, 0]      #"other-vehicle"
  5 : [30, 30, 255]    #"person"
  6 : [200, 40, 255]   #"bicyclist"
  7 : [90, 30, 150]    #"motorcyclist"
  8 : [255, 0, 255]    #"road"
  9 : [255, 150, 255]  #"parking"
  10: [75, 0, 75]      #"sidewalk"
  11: [75, 0, 175]     #"other-ground"
  12: [0, 200, 255]    #"building"
  13: [50, 120, 255]   #"fence"
  14: [0, 175, 0]      #"vegetation"
  15: [0, 60, 135]     #"trunk"
  16: [80, 240, 150]   #"terrain"
  17: [150, 240, 255]  #"pole"
  18: [0, 0, 255]      #"traffic-sign"